app/models/ability.rb

```diff
@@ -0,0 +1,20 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    user ||= User.new # guest user
+    can :read, Post, public: true
+
+    return unless user.present?
+
+    can :read, Post, user_id: user.id
+    can :destroy, Post, author_id: user.id
+    can :destroy, Comment, author_id: user.id
+    can :destroy, Post, id: user.posts.pluck(:id)
+
+    return unless user.admin?
+
+    can :manage, Post
+    can :manage, Comment
+  end
+end
```

Comment:
- [ ] Great work implementing the suggested changes from the previous reviewer. After creating a user to test your app, I got below error message while trying to create a post as a new user. Kinldy look into the auth rule and readdress the rule.

### App Behaviour
[error.webm](https://github.com/Santosh-Konappanavar/Blog_App/assets/26597432/eaae50cf-c671-4abe-b3f0-99c5b1fa566a)

### Error
![post-error](https://github.com/Santosh-Konappanavar/Blog_App/assets/26597432/0222175e-7c49-425b-a498-ba331fe2befb)

You can check this for insight

```
   class Ability
 include CanCan::Ability

     def initialize(user)

         user ||= User.new # guest user (not logged in)

         if user.admin? 
         can :manage, :all
         else
         can :manage, Post, author: user  # owner of post can perform all view, update or destroy own post if Post.author == user
         can :manage, Comment, author: user
         can :read, :all
         end
     end
 end

```